accord-core/src/main/java/accord/impl/SimpleProgressLog.java

```diff
@@ -375,6 +377,8 @@ public String toString()
         void recordBlocking(Command blocking, Keys someKeys)
         {
             this.blocking = blocking;
+            if (blocking.txn() != null && !blocking.txn().keys.containsAll(someKeys))
+                throw new IllegalStateException();
```

Comment:
would be good to produce a meaningful error, so if this ever happens we have enough to debug

Reply:
I think this will only come up in BurnTests where it's easily reproducible, and I think this line is being removed anyway in the next patch. But I'll add some for now anyway.